test/e2e/dra/deploy.go

```diff
@@ -314,7 +313,7 @@ func (d *Driver) SetUp(nodes *Nodes, resources app.Resources) {
 		nodename := pod.Spec.NodeName
 		logger := klog.LoggerWithValues(klog.LoggerWithName(klog.Background(), "kubelet plugin"), "node", pod.Spec.NodeName, "pod", klog.KObj(&pod))
 		loggerCtx := klog.NewContext(ctx, logger)
-		plugin, err := app.StartPlugin(loggerCtx, "/cdi", d.Name, nodename,
+		plugin, err := app.StartPlugin(loggerCtx, "/cdi", d.Name, d.f.ClientSet, nodename,
```

Comment:
Am I reading correctly that this is running instances of the example plugin from the control plane with the e2e test client, which is typically a superuser?

I think this should set up scoped clients matching how we'd actually expect these to run on nodes, with granted permissions and guarded with validating admission to demonstrate that is possible and how to do it

Normally, I guess these would run in daemonset pods and automatically get node-scoped token credentials?

You can mock this up in an e2e by impersonating the node-scoped service account the driver would be running as, something like this:

```go
import "k8s.io/apiserver/pkg/authentication/serviceaccount"

...

		driverUserInfo := (&serviceaccount.ServiceAccountInfo{
			Name:      "example-driver",
			Namespace: "example-namespace",
			NodeName:  nodename,
		}).UserInfo()
		driverClientConfig := d.f.ClientConfig()
		driverClientConfig.Impersonate = rest.ImpersonationConfig{
			UserName: driverUserInfo.GetName(),
			Groups:   driverUserInfo.GetGroups(),
			Extra:    driverUserInfo.GetExtra(),
		}
		driverClient, err := clientset.NewForConfig(driverClientConfig)
```

Then you'd need to set up a single RBAC role / binding granting the driverUserInfo.GetName() user permissions, and could set up an example Validating Admission Policy showing how to guard writes to those resources for that user based on the node name extra info

Reply:
That sounds like a good idea. I'm not sure when I'll get to it, though. I'll either have to keep this PR open until I have the time, or add a TODO + issue for adding the VAP later. Any preference?

Reply:
I don't want to let the driver start running with a superuser client ... that's an almost guaranteed way for spurious API calls to get in and start being relied on

Reply:
That part is easy to add and definitely something that I want to do now. The VAP part scares me because I don't know it and the example seemed not at all simple.

Reply:
I added impersonation (see last commit), but not VAP.

Reply:
Tracking issue for VAP: https://github.com/kubernetes-sigs/dra-example-driver/issues/48